api/v1beta2/network_types.go

```diff
@@ -455,6 +455,27 @@ type VPCSpec struct {
 	// +optional
 	// +kubebuilder:validation:Enum:=ip-name;resource-name
 	PrivateDNSHostnameTypeOnLaunch *string `json:"privateDnsHostnameTypeOnLaunch,omitempty"`
+
+	// PublicIpv4Pool defines the Public IPv4 Pool to be used for VPC resources created
+	// in public subnets.
+	//
+	// +kubebuilder:validation:MaxLength=30
+	// +kubebuilder:validation:XValidation:rule="self.startsWith('ipv4pool-ec2-')",message="publicIpv4Pool must start with 'ipv4pool-ec2-'"
+	// +optional
+	PublicIpv4Pool *string `json:"publicIpv4Pool,omitempty"`
+
+	// PublicIpv4PoolFallBackOrder defines the fallback when the Public IPv4 Pool has been exhausted,
+	// no more IPv4 address available in the pool. NOTE: this is not implemented yet.
```

Comment:
> NOTE: this is not implemented yet.

I believe this is reference in other areas of the code.  I believe only none is supported at the moment, is that correct?

Reply:
Good catch. We can remove this note. This vesion implements the fallback [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4905/files#diff-a3d373b4a7875fa24f1be6f3b2ab92a57fca9d7928388e7d0e1f782a6374706dR61-R87)